### ISSUE TITLE
BUG: SARIMAX numpy compatibility

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2798]
Looks like conda in TravisCI testing updated numpy, which shows a new `IndexError`
numpy:           1.10.4-py27_0 

https://travis-ci.org/statsmodels/statsmodels/jobs/107300505

```
======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_sarimax.Test_arma_exog_trend_polynomial_missing.test_results
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/tsa/statespace/tests/test_sarimax.py", line 922, in test_results
    self.result.summary()
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/tsa/statespace/sarimax.py", line 2031, in summary
    alpha=alpha, start=start, model_name=model_name
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/tsa/statespace/mlemodel.py", line 2183, in summary
    diagn_right = [('Jarque-Bera (JB):', format_str(jb[:,0])),
IndexError: too many indices for array
```

**Comment by josef-pkt** [comment]
@ChadFulton  Can you check the dimension of the Jarque-Bera return in SARIMAX?

**Comment by thequackdaddy** [comment]
Hello,

I have a pull request that fixes this bug. 

https://github.com/statsmodels/statsmodels/pull/2800

However, there appears to be an MKL bug which is still outstanding. 

Thanks!

**Comment by josef-pkt** [comment]
I'm pretty sure that the MKL problem is a bug in conda (mixing fortran compilers). 
They switched from openblas to mkl two days ago, and this was working before. I'll wait if conda gets fixed before trying to figure out the details

**Comment by ChadFulton** [comment]
Fixed in #2809 (forgot to do the auto-close thing again ;)
